server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1435,22 +1439,28 @@ private void closeConsistencyCheck() {
     }
   }
 
-  private void compareToDataInMemory(TabletMetadata tabletMetadata) {
-    if (!tabletMetadata.getFilesMap().equals(getDatafileManager().getDatafileSizes())) {
-      String msg = "Data files in " + extent + " differ from in-memory data "
-          + tabletMetadata.getFilesMap() + " " + getDatafileManager().getDatafileSizes();
-      log.error(msg);
+  public synchronized void compareTabletInfo(MetadataUpdateCount updateCounter,
+      TabletMetadata tabletMetadata) {
+    if (isClosed() || isClosing()) {
+      return;
     }
-  }
 
-  public synchronized void compareTabletInfo(Long updateCounter, TabletMetadata tabletMetadata) {
-    if (isClosed() || isClosing()) {
+    if (updateCounter.overlapsUpdate() || !updateCounter.equals(this.getUpdateCount())) {
+      // for these cases do not even bother checking if the files are the same
       return;
     }
```

Comment:
:+1: for the first case, where we're checking while an update is occurring. But, I'm not sure I understand the second case. What situation does that skip?

Reply:
In the second case there is evidence that a change has occurred, so its an optimization to avoid comparing the maps.

Reply:
> In the second case there is evidence that a change has occurred, so its an optimization to avoid comparing the maps.

Isn't this evidence that either:
1. a change has occurred, or
2. a possible inconsistency has been detected?

Reply:
If a change has occurred then we can reliably check for inconsistencies, so need to rely on the fact that it will eventually check again later.

Reply:
The big picture is that we are periodically reading a tablets metadata from the metadata table and comparing that to what the tablet has in memory.  We only want to do this comparison when there were no changes to the tablet while were in the process of reading the metadata table.   So when we know for sure a change has occurred to the tablet while we were reading from the metadata table there is no need to compare what we got from the metadata table to the tablets current files.

Reply:
I misunderstood the object returned from `getUpdateCount()` as still being a primitive, and not the new MetadataUpdateCount type you created. It makes sense now. Thanks.